### PR TITLE
account: stub ListQualifiedUsers

### PR DESCRIPTION
**Pull request by makigumo**
Allows to start:
* _Contra: Rogue Corps_
* _Bulletstorm: Duke of Switch Edition_

### CODE REVIEW

File: src/core/hle/service/acc/acc.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include "core/hle/service/glue/manager.h"
 #include "core/hle/service/sm/sm.h"
 #include "core/loader/loader.h"
+using namespace Common;
```
Comment (FearlessTobi) [inline src/core/hle/service/acc/acc.cpp]: Don't do that.

File: src/core/hle/service/acc/acc.cpp [hunk 2]
```diff
@@ -340,6 +341,13 @@ void Module::Interface::InitializeApplicationInfoRestricted(Kernel::HLERequestCo
     rb.Push(InitializeApplicationInfoBase(pid));
 }
 
+void Module::Interface::ListQualifiedUsers(Kernel::HLERequestContext& ctx) {
+    LOG_DEBUG(Service_ACC, "(STUBBED) called");
```
Comment (FearlessTobi) [inline src/core/hle/service/acc/acc.cpp]: LOG_WARNING

**Comment by makigumo** [comment]
Thanks, fixed.

This is basically guesswork from the workings of the other functions. I haven't really gotten far in my RE efforts yet :(

File: src/core/hle/service/acc/acc.cpp [hunk 3]
```diff
@@ -340,6 +340,13 @@ void Module::Interface::InitializeApplicationInfoRestricted(Kernel::HLERequestCo
     rb.Push(InitializeApplicationInfoBase(pid));
 }
 
+void Module::Interface::ListQualifiedUsers(Kernel::HLERequestContext& ctx) {
+    LOG_WARNING(Service_ACC, "(STUBBED) called");
+    ctx.WriteBuffer(profile_manager->GetAllUsers());
```
Comment (lioncash) [inline src/core/hle/service/acc/acc.cpp]: I'm quite sure this isn't correct based off what I remember from RE a while ago. Qualified users can differ from the total list of users if I'm not mistaken.

Comment (CJBok) [reply src/core/hle/service/acc/acc.cpp]: This is purely a guess, but I think ListQualifiedUsers might be related to Parental Control.
I know that Contra Rogue Corps which is Rated M and uses this function at start of the game.

Comment (theboy181) [reply src/core/hle/service/acc/acc.cpp:345]: What is the Parental Control function called ?

Comment (CJBok) [reply src/core/hle/service/acc/acc.cpp:345]: It is a configuration setting on the Switch itself. To be more precise, it is possibly related to Parental Control -> Restrictred Software where you can set the minimum age of which a profile is not allowed to play. So possibly the function contains a minimum age parameter unless this information is already given in a different function.

image

**Pull request closed without merging**